pkg/services/util/sign.go

```diff
@@ -229,7 +58,11 @@ var (
 // ToStatus unwraps the deepest error from err and converts it into the response
 // status.
 func ToStatus(err error) *protostatus.Status {
-	return statusFromErr(err).ToGRPCMessage().(*protostatus.Status)
+	// unwrap error
+	for e := errors.Unwrap(err); e != nil; e = errors.Unwrap(err) {
```

Comment:
I see it's the old code inlined, but this doesn't seem to be correct. It implies that the most inner error is `apistatus.*` one, but this assumption can be wrong. The way `ErrorToV2()` is implemented (using `errors.As()`) I'd say this is not needed.

Reply:
we have #2744 bout this, or i changed the behavior?